utils.py

```diff
@@ -43,7 +47,74 @@ def server_response_decode(message):
             remainder = message_parts[1][b_string_len + 2 :]
 
     elif message[0] == ":":  # responses should start with '(integer)'
-        print("integer")
-        value = int(message_parts[0])
+        value = f"(integer) {message_parts[0]}"
         remainder = message_parts[1]
     return value, remainder
+
+
+def generate_redis_formatted_array(key_or_value, cli_message):
+    # Generates start of array taking length of list based on number of arguments passed to it.
+    command_length = len(cli_message)
+    total_length = 1 + len(key_or_value)
+    encoded_message_start = f"{ARRAY_START}{total_length}{SEPARATOR}{BULK_STR_START}{command_length}{SEPARATOR}{cli_message}{SEPARATOR}"
+
+    # Creates and returns redis formatted array to send to Redis server
+    for key in key_or_value:
+        encoded_message_start += (
+            f"{BULK_STR_START}{len(key)}{SEPARATOR}{key}{SEPARATOR}"
+        )
+    return encoded_message_start
+
+
+def command_handler(message, dict):
+    # generates bulk string, simple string, null reply and integer reply.
+    if type(message) != list:
+        return f"-ERR unkown command {*message,}{SEPARATOR}"
+    if message[0] == "GET":
+        if len(message) > 2:
+            return f"-ERR unkown command {*message[2:],}{SEPARATOR}"
+        elif message[1] not in dict:
+            return NIL_REPLY
+        elif message[1] in dict:
+            value_len = len(dict[message[1]])
+            return f"${value_len}{SEPARATOR}{dict[message[1]]}{SEPARATOR}"
+    if message[0] == "DEL":
+        count = 0
+        del_len = len(message) - 1
+        for i in range(del_len):
+            if message[i + 1] in dict:
+                del dict[message[i + 1]]
+                count += 1
+        return f":{count}{SEPARATOR}"
+    if message[0] == "SET":
+        if len(message) >= 4:
+            if message[3] == "GET":
+                if message[1] in dict:
+                    old_key_len = len(dict[message[1]])
+                    old_key = dict[message[1]]
+                    old_key_BS = f"${old_key_len}{SEPARATOR}{old_key}{SEPARATOR}"
+                    dict.update({message[1]: message[2]})
+                    return old_key_BS
+                else:
```

Comment:
can be removed because of `return` above

Reply:
fixed